src/fsharp/NameResolution.fs

```diff
@@ -1498,15 +1498,47 @@ type TcSymbolUseData =
 /// are allocated (one per file), but they are large because the allUsesOfAllSymbols array is large.
 type TcSymbolUses(g, capturedNameResolutions : ResizeArray<CapturedNameResolution>, formatSpecifierLocations: (range * int)[]) = 
     
+    let chunks (items: ResizeArray<'t>) chunkSize =
+        let itemCount = items.Count
+        let chunksCount =
+           match itemCount / chunkSize with
+           | n when itemCount % chunkSize = 0 -> n
+           | n -> n + 1 // any remainder means we need an additional chunk to store it
+
+        [| for index in 0..chunksCount-1 do
+               let startIndex = index * chunkSize
+               let takeCount =
```

Comment:
Is this `let takeCount = min (itemCount - startIndex) chunkSize`?

Reply:
Yes it is, thanks for the catch.